projects/core/src/lib/layers/symbology/raster-symbology-editor/raster-symbology-editor.component.ts

```diff
@@ -94,35 +114,52 @@ export class RasterSymbologyEditorComponent implements OnInit {
     }
 
     getColorizerType(): ColorizerType {
-        if (this.symbology.colorizer instanceof LinearGradient) {
+        const colorizer = this.getActualColorizer();
+
+        if (colorizer instanceof LinearGradient) {
             return LinearGradient.TYPE_NAME;
         }
 
-        if (this.symbology.colorizer instanceof PaletteColorizer) {
+        if (colorizer instanceof PaletteColorizer) {
             return PaletteColorizer.TYPE_NAME;
         }
 
-        if (this.symbology.colorizer instanceof LogarithmicGradient) {
+        if (colorizer instanceof LogarithmicGradient) {
             return LogarithmicGradient.TYPE_NAME;
         }
 
-        if (this.symbology.colorizer instanceof RgbaColorizer) {
+        if (colorizer instanceof RgbaColorizer) {
             return RgbaColorizer.TYPE_NAME;
         }
 
         throw Error('unknown colorizer type');
     }
 
+    setSelectedBand(band: RasterBandDescriptor): void {
+        this.selectedBand = band;
+        if (this.symbology.colorizer instanceof SingleBandRasterColorizer) {
```

Comment:
what if it is another type?

Reply:
Currently, it can't be another type, because no other RasterColorizer type exists.
Once we add a new type (Rgba) we need to implement the logic for managing it in the UI. Then we should have two subcomponents for the raster symbology editor, one for single-band, and one for rgba. And then we need a selector for choosing the type of raster colorizer and ensure there that we show the user the correct symbology editor that can then be assured to get a matching symbology with the matching raster colorizer as input.